src/crimson/os/seastore/transaction_manager.h

```diff
@@ -1031,36 +1002,34 @@ class TransactionManager : public ExtentCallbackInterface {
   template <typename T>
   pin_to_extent_ret<T> pin_to_extent(
     Transaction &t,
-    LBAMappingRef pin,
+    LBAMapping pin,
     extent_len_t direct_partial_off,
     extent_len_t partial_len) {
     static_assert(is_logical_type(T::TYPE));
     using ret = pin_to_extent_ret<T>;
-    auto &pref = *pin;
-    auto direct_length = pref.is_indirect() ?
-      pref.get_intermediate_length() :
-      pref.get_length();
+    auto direct_length = pin.is_indirect() ?
+      pin.get_intermediate_length() :
+      pin.get_length();
     if (full_extent_integrity_check) {
       direct_partial_off = 0;
       partial_len = direct_length;
     }
     LOG_PREFIX(TransactionManager::pin_to_extent);
     SUBTRACET(seastore_tm, "getting absent extent from pin {}, 0x{:x}~0x{:x} ...",
-              t, *pin, direct_partial_off, partial_len);
+              t, pin, direct_partial_off, partial_len);
     return cache->get_absent_extent<T>(
       t,
-      pref.get_val(),
+      pin.get_val(),
       direct_length,
       direct_partial_off,
       partial_len,
-      [&pref]
+      [&pin]
       (T &extent) mutable {
 	assert(!extent.has_laddr());
 	assert(!extent.has_been_invalidated());
-	assert(!pref.has_been_invalidated());
-	assert(pref.get_parent());
-	pref.link_child(&extent);
-	extent.maybe_set_intermediate_laddr(pref);
+	assert(pin.is_valid());
+	pin.link_child(extent);
+	extent.maybe_set_intermediate_laddr(pin);
```

Comment:
Same here, I think we can directly set the extent laddr from `get_intermediate_base()` regardlessly.